functions/contrib/go/annotate-apply-time-mutations/README.md

```diff
@@ -0,0 +1,101 @@
+# annotate-apply-time-mutations
+
+## Overview
+
+<!--mdtogo:Short-->
+
+The `annotate-apply-time-mutations` function enables authors to use inline comments,
+rather than annotations, to specify field replacements using the apply time mutation feature.
+
+It works by reading `# apply-time-mutation:` comments on resource YAML and adds the equivalent
+`config.kubernetes.io/apply-time-mutation` annotation to the resource.
+
+The `config.kubernetes.io/apply-time-mutation` annotation is read by the [apply time mutation]
+functionality which patches the resource config at apply time, during `kpt live apply`, with
+the referenced resource's live value.
+
+<!--mdtogo-->
+
+<!--mdtogo:Long-->
+
+## Usage
+
+The annotate-apply-time-mutations function can be executed declaratively as part of `kpt fn render`
+
+```yaml
+apiVersion: kpt.dev/v1
+kind: Kptfile
+pipeline:
+  mutators:
+    - image: gcr.io/kpt-fn-contrib/annotate-apply-time-mutations:unstable
+```
+
+or imperatively like:
+
+```shell
+kpt fn eval --image gcr.io/kpt-fn-contrib/annotate-apply-time-mutations:unstable
+```
+
+
+The `annotate-apply-time-mutations` function does the following:
+
+1.  Scans the package for `apply-time-mutation` comment markup.
+2.  Appends the equivalent `config.k8s.io/apply-time-mutation` annotation to the same.
+
+The expected `apply-time-mutation` comment format is:
+
+`# apply-time-mutation: "[prefix]${[group]/[version]/namespaces/[source namespace]/[kind]/[source name]:[source field path]}[suffix]`
```

Comment:
> [group]/[version]/namespaces/[source namespace]/[kind]/[source name]

It seems it's similar to k8s resources path, but it's slightly different. In k8s APIServer it's `[group]/[version]/namespaces/[namespace]/[resources]/[name]`. e.g. `apps/v1/namespaces/kube-system/deployments/coredns` in k8s. It'd be `apps/v1/namespaces/kube-system/Deployment/coredns` in your case. Is that expected?

Reply:
Do you mean actually `resource namespace` by using `source namespace`?
Same question for `source name`.

Reply:
this resource addressing syntax is matched to the depends-on annotation implementation for consistency